std/cpp/Lib.hx

```diff
@@ -27,55 +28,69 @@ package cpp;
 	and vice-versa.
 **/
 class Lib {
-
 	/**
 		Load and return a Cpp primitive from a DLL library.
 	**/
-	public static function load( lib : String, prim : String, nargs : Int ) : Dynamic {
+	public static function load(lib:String, prim:String, nargs:Int):Dynamic {
 		#if (iphone || emscripten)
-		return loadLazy(lib,prim,nargs);
+		return loadLazy(lib, prim, nargs);
 		#else
-		return untyped __global__.__loadprim(lib,prim,nargs);
+		return untyped __global__.__loadprim(lib, prim, nargs);
 		#end
 	}
 
 	/**
 		Unloaded all dynamic libraries in reverse order of loading.
 		Returns the number of libraries unloaded.
 	**/
-	public static function unloadAllLibraries() : Int {
+	public static function unloadAllLibraries():Int {
 		return untyped __global__.__hxcpp_unload_all_libraries();
 	}
 
-	public static function _loadPrime( lib : String, prim : String, signature : String, quietFail = false ) : Dynamic {
-		var factory:Callable< ConstCharStar -> Object > =
-		untyped __global__.__hxcpp_cast_get_proc_address(lib, prim + "__prime", quietFail);
-		if (factory!=null)
-		{
+	public static function _loadPrime(lib:String, prim:String, signature:String, quietFail = false):Dynamic {
+		var factory:Callable<ConstCharStar->Object> = untyped __global__.__hxcpp_cast_get_proc_address(lib, prim + "__prime", quietFail);
+		if (factory != null) {
 			var func:Dynamic = factory.call(signature);
-			if (func==null && !quietFail)
+			if (func == null && !quietFail)
 				throw '$prim does not have signature $signature';
 			return func;
 		}
 		return null;
 	}
 
-
 	/**
 		Tries to load, and always returns a valid function, but the function may throw
 		if called.
 	**/
-	public static function loadLazy(lib:String, prim:String, nargs:Int) : Dynamic {
+	public static function loadLazy(lib:String, prim:String, nargs:Int):Dynamic {
 		try {
-			return untyped __global__.__loadprim(lib,prim,nargs);
-		} catch( e : Dynamic ) {
-			switch(nargs) {
-			case 0 : return function() { throw e; };
-			case 2 : return function(_1,_2) { throw e; };
-			case 3 : return function(_1,_2,_3) { throw e; };
-			case 4 : return function(_1,_2,_3,_4) { throw e; };
-			case 5 : return function(_1,_2,_3,_4,_5) { throw e; };
-			default : return function(_1) { throw e; };
+			return untyped __global__.__loadprim(lib, prim, nargs);
+		} catch (e:Dynamic) {
+			switch (nargs) {
+				case 0:
+					return function() {
+						throw e;
+					};
+				case 2:
+					return function(_1, _2) {
+						throw e;
+					};
+				case 3:
+					return function(_1, _2, _3) {
+						throw e;
+					};
+				case 4:
+					return function(_1, _2, _3, _4) {
+						throw e;
+					};
+				case 5:
+					return function(_1, _2, _3, _4, _5) {
+						throw e;
+					};
+				default:
+					return function(_1) {
+						throw e;
+					};
```

Comment:
Something need to be done about this too.

Reply:
What the formatter does with it seems logical to me. However, to make it a lot more concise, it can be:
- changed to a `return switch` instead of having returns in every branch - this will keep the case bodies on the same line
- changed the functions to arrow functions

Reply:
What exactly do you want here, @ncannasse ? Keeping the `throw` on the same line? I think putting the `return` out of the `switch`, then changing the functions to e.g. `(_1, _2) -> throw e;` should be good.

Reply:
While we're at it, what's with the awkward `_<number>` naming? You can have multiple args named `_`.